src/main/java/com/mathworks/ci/RunMatlabTestsBuilder.java

```diff
@@ -284,6 +287,12 @@ private String getInputArguments() {
             artifact.addFilePathArgTo(args);
         }
 
+        // Add source folder options to argument

```

Comment:
I wonder if this should follow the existing Artifact to arguments pattern instead of standing alone like this. It could be that this particular artifact may need custom behavior when gets converted to arguments.

Reply:
There's a slight difference in notion of an Artifact in this class and Source Folder option, that is the Artifact is generated and Source Folder is a configuration for running test and collecting coverage. Moreover, the implementation for Source Folder differs from Artifact as for it (source folder) we need a describable class. If the source folder variable is initialized with any object, it will show up as Selected in UI (because of optionalProperty jelly tag) hence a Null Object Pattern probably wont be fit for this.